rfd/0180-sso-as-mfa-method.md

```diff
@@ -0,0 +1,528 @@
+from registering/using MFA devices registered through Teleport.
+
+#### Default MFA connector
+
+Cluster auth preference will also gain the `mfa_connector_name` field to set
+a preferred IdP connector for MFA checks. As detailed in the UX section, setting
+a preferred IdP connector has some beneficial UX implications.
+
+If `mfa_connector_name` is not set, but `connector_name` is set, that connector
+will be used as an MFA method if `allow_as_mfa_method` is set to `yes`.
+
+If neither of the fields above are set, Teleport will look through all
+registered connectors in lexical order and return the first one with 
+`allow_as_mfa_method` set to `yes` or`only`.
+
+#### Bad configuration
+
+The following two fail states should be prevented:
+- `mfa_connector_name` points to a connector with `allow_as_mfa_method = no`
+- `connector_name` points to a connector with `allow_as_mfa_method = only`
+
+These fail states will be checked on both connector update and auth preference
+update.
+
+### Security
+
+Teleport uses MFA checks for some of its most security focused features, including
+per-session MFA, moderated sessions, and MFA for admin actions. Using SSO as an
+MFA method opens up the possibility of poorly configured clusters being
+vulnerable to attacks ranging from internal users avoiding safe MFA checks to
+attackers with a compromised IdP gaining keys to the castle.
+
+#### Opt-in
+
+SSO as an MFA method will be opt-in. Administrators will be instructed through
+the docs to only enable an IdP connector as an MFA method if the IdP provider
+has strict checks itself (e.g. Administered Webauthn devices, Trusted devices).
+
+Teleport has no way to confirm whether a registered IdP connector follows the
+guidelines, but it will display a warning to admins who attempt to enable it.
+
+```console
+> tctl edit connector/okta
+### sets `allow_as_mfa_method: yes`
+Warning: Allowing this IdP provider to be used as an MFA method may reduce the
+security of enforced MFA checks for critical security features. This option
+should not be enabled unless the IdP provider has strict MFA and/or Device trust
+enforcement itself. Continue? (y/N):
+```
+
+#### IdP Compromise
+
+In the case of a full-scale IdP compromise, an attacker may have the ability
+to auto-provision users with arbitrary permissions.
+
+When device trust is required, newly auto-provisioned SSO users are required
+to add their first MFA device from a trusted device. When combined with MFA
+security features, such as MFA for Admin actions and per-session MFA, the blast
+radius of an IdP compromise is largely contained. The attacker would be
+prevented from accessing any critical infrastructure or making any changes
+to the cluster's security configuration.
+
+Allowing SSO as an MFA method would bypass the device trust check, opening the
+cluster back up to attacks in the case of an IdP compromise. To maintain this
+invariant, device trust must be enforced within the SSO MFA check.
```

Comment:
> To maintain this invariant, device trust must be enforced within the SSO MFA check.

Does this mean it's automatically enforced by Teleport in this situation? Just to clarify.

Reply:
If device trust is enabled, yes. I'm not sure if this is already the case with SSO login flow.

Reply:
Ack. Thanks for clarifying.